### PR TITLE
feature: adds new event triggers to items

### PR DESCRIPTION
**Pull request by AlexVild**
Resolves #2181 

Adds the following options to the Item editor:

image

Have tested these and they all work to my knowledge.

### CODE REVIEW

**Review by Mighty-Professional — DISMISSED** [review]

**Review by Mighty-Professional — APPROVED** [review]

**Review by WeylonSantana — CHANGES_REQUESTED** [review]

**Comment by WeylonSantana** [comment]
I have one more extra thing to say, the selection is very strange in the editor, especially if you intend to use the mouse scroll


https://github.com/AscensionGameDev/Intersect-Engine/assets/63019821/7cdc6e9c-67b4-4755-b972-7921928492b0

File: Intersect (Core)/GameObjects/ItemBase.cs [hunk 1]
```diff
@@ -342,6 +342,26 @@ public string EffectsJson
         [NotMapped, JsonIgnore]
         public ItemRange[] StatRanges => EquipmentProperties?.StatRanges?.Values.ToArray();
 
+        [Column(nameof(EventTriggers))]
+        public string EventTriggersJson
+        {
+            get => JsonConvert.SerializeObject(EventTriggers);
+            set => EventTriggers = JsonConvert.DeserializeObject<Dictionary<ItemEventTriggers, Guid>>(value ?? "") ?? new Dictionary<ItemEventTriggers, Guid>();
```
[the remark this answers is not in the document]
Comment (AlexVild) [reply Intersect (Core)/GameObjects/ItemBase.cs]: fixed

**Comment by AlexVild** [comment]
@WeylonSantana Fixed the editor issue

**Review by WeylonSantana — APPROVED** [review]

File: Intersect.Server.Core/Entities/Entity.cs [hunk 2]
```diff
@@ -2169,21 +2175,21 @@ public virtual void TryAttack(Entity target)
...
                     Animate(enemy, aliveAnimations);
                 }
-
-                //Check for any onhit damage bonus effects!
-                CheckForOnhitAttack(enemy, isAutoAttack);
             }
 
+            //Check for any onhit damage bonus effects!
+            CheckForOnhitAttack(enemy, isAutoAttack);
+
```
Comment (Arufonsu) [inline Intersect.Server.Core/Entities/Entity.cs:2182]: Hello, gave this PR a try and it seems good so far, just wanted to know though, why was this one call changed outside the previous conditional branches?
as for the rest, tested with a pair of items, so far seems to behave as expected, approved 👍🏽

**Review by Arufonsu — APPROVED** [review]

**Pull request merged**